### PR TITLE
fix glooctl demo federation

### PR DESCRIPTION
**Pull request by ashleywang1**
# Description

Gloo has been changed to pass in the local KubeCfg when registering and deregistering clusters.

# Context

glooctl demo federation does not work for glooctl version v1.6.0-beta16 to v1.6.0-beta19

In Gloo v1.6.0-beta16, we pulled in a skv2 dependency change from v0.8.1 to v0.13.5.

This change brings with it several changes to the way we register clusters (specifically for Gloo Fed). We now create the KubernetesCluster CRD on the management cluster, which requires us to have the mgmtKubeCfg passed into the registration options. Our Gloo code does not pass in a KubeCfg, resulting in a segfault. This has been fixed.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3972

### CODE REVIEW

**Comment by ashleywang1** [comment]
Testing was done in gloo-fed, by running this script:
```
make TAGGED_VERSION=v1.6.0-beta20 glooctl-darwin-amd64 -B; mv _output/glooctl-darwin-amd64 _output/glooctl; export PATH=_output:$PATH
glooctl demo federation --license-key=$LICENSE_KEY
```

And checking that gloo-fed worked as expected.

File: projects/gloo/cli/pkg/cmd/federation/register/root.go [hunk 1]
```diff
@@ -27,6 +27,14 @@ var glooFederationPolicyRules = []v1.PolicyRule{
 	},
 }
 
+var glooFederationReadConfigPolicyRules = []v1.PolicyRule{
```
Comment (ashleywang1) [inline projects/gloo/cli/pkg/cmd/federation/register/root.go:30]: This is part of the solution for https://github.com/solo-io/gloo-fed/issues/281

File: go.mod [hunk 2]
```diff
@@ -124,6 +124,7 @@ replace (
 	github.com/census-instrumentation/opencensus-proto => github.com/census-instrumentation/opencensus-proto v0.2.0 // indirect
 	github.com/docker/docker => github.com/moby/moby v0.7.3-0.20190826074503-38ab9da00309
 	// Required for proper serialization of CRDs
+	github.com/golang/protobuf => github.com/golang/protobuf v1.4.3
```
Comment (kdorosh) [inline go.mod:127]: why add this?

Comment (ashleywang1) [reply go.mod:127]: Ah, I ran into an error before that was only fixed by adding this, but I've taken this line out and the code compiles just fine.

**Review by kdorosh — APPROVED** [review]

**Pull request merged**